File: src/main/java/ladder/domain/LEVEL.java
package ladder.domain;

import lombok.ToString;

@ToString
public enum LEVEL {
    HIGH("상", 0.8, 20),
    MIDDEL("중", 0.5, 10),
    LOW("하", 0.2, 5);

    private String level;
    private double seed;
    private int lineSize;

    LEVEL(String level, double seed, int lineSize) {
        this.level = level;
        this.seed = seed;
        this.lineSize = lineSize;
    }

    public static LEVEL findLevel(String searchLevel) {
        for (LEVEL value : LEVEL.values()) {
            if(value.level.equals(searchLevel)) {
                return value;
            }
        }
        throw new IllegalArgumentException("잘못 입력했지?!");
    }

    public int getLineSize() {
        return lineSize;
    }

    public double getSeed() {
        return seed;
    }
}

File: src/main/kotlin/ladder/domain/Level.kt
package ladder.domain

enum class Level(private val levelName: String,
                 private val height: Int,
                 private val percent: Float) {
    HIGH("상", 20, 0.8F),
    MEDIUM("중", 10, 0.5F),
    LOW("하", 5, 0.3F);

    fun getPercent(): Float = percent

    fun getHeight(): Int = height

    companion object {
        fun findLevel(levelName: String): Level =
                values()
                        .find { it.levelName == levelName }
                        ?: throw IllegalArgumentException("존재하지 않는 난이도입니다.")
    }
}
